### ISSUE TITLE
"request" in worker.events.on('response:mocked') is always null

### ISSUE THREAD

**Issue by kettanaito** [kettanaito/msw#1992]
### Discussed in https://github.com/mswjs/msw/discussions/1991

<div type='discussions-op-text'>

<sup>Originally posted by **canrozanes** January 22, 2024</sup>
Hi, 

I was following the docs on [life-cycle events](https://mswjs.io/docs/api/life-cycle-events/) and I was trying to assert on a response after making an API call.

I have the following piece of code that saves all the requests so that I can assert on the responses later.

```
// QUnit  setup
hooks.beforeEach(async function (this: TestContextWithMsw) {
    this.worker = setupWorker(...handlers);
    this.worker.events.on('response:mocked', ({ request, requestId, response }) => {
      this.handledRequests.push({ request, requestId, response });
    });
});
```

I can clearly see that MSW sends a response in the browser network tab but when I add a debugger inside the 'response:mocked' callback, I see that all the request are null. 

![Screenshot 2024-01-22 at 10 39 35 AM](https://github.com/mswjs/msw/assets/37912128/0fe4ceba-aedb-4a47-a3b3-581b414e1e4c)

What could I be missing?</div>

**Comment by kettanaito** [comment]
I'd start here: 

https://github.com/mswjs/msw/blob/396b12ffd084115c22e9df10a4f6fd59ff8c5a70/src/browser/setupWorker/start/createResponseListener.ts#L49-L53

And figure out why we don't know anything about the request.

**Comment by kettanaito** [comment]
Opened a fix at #1993

**Comment by kettanaito** [comment]
## Released: v2.1.5 🎉

This has been released in v2.1.5!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v2.1.5)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/2.1.5)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
